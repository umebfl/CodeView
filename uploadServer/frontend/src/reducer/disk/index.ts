import {
    addIndex,
    find,
    findIndex,
    keys,
    map,
    propEq,
    reduce,
    take,
    update,
} from 'ramda'
import { createModel } from '@rematch/core'

import request from 'src/util/request'

import { RootModel } from '..'
import { DiskType, diskState, DiskResponseType } from 'src/reducer/disk/type'
import { DEFAULT_DATA_SOURCE } from 'src/reducer/userConfig'

const MAX_COMMENT_LEN = 120

const initState: diskState = {
    data: [],
    uploadRecords: {},
}

export const disk = createModel<RootModel>()({
    state: initState,
    reducers: {
        setData: (state, payload: Partial<diskState>) => {
            return {
                ...state,
                ...payload,
            }
        },
    },
    effects: dispatch => ({
        async initData(_, rootState) {
            const { dataSource } = rootState.userConfig

            const ds = dataSource || DEFAULT_DATA_SOURCE

            const data: {
                disks_info: DiskResponseType[]
            } = await request({
                url: `/disk_management/get_disks_info?region=${ds}`,
                rootState,
                dispatch,
            })

            const { disks_info } = data

            if (disks_info) {
                const transData: DiskType[] = addIndex<DiskResponseType>(map)(
                    (disk: DiskResponseType, idx: number): DiskType => ({
                        id: idx,
                        diskId: disk.disk_sn,
                        owner: disk.owner,
                        inventoryStatus: disk.status,
                        onServer: disk.on_server,
                        comment: disk.comment,
                    })
                )(disks_info)

                dispatch.disk.setData({
                    data: transData,
                })
            }
        },

        async getUploadRecords(payload: string, rootState) {
            try {
                const data = await request({
                    url: `/disk_management/get_upload_records?disk_sn=${payload}`,
                    rootState,
                    dispatch,
                })
                const { disk } = rootState

                if (data.upload_records) {
                    dispatch.disk.setData({
                        uploadRecords: {
                            ...disk.uploadRecords,
                            [payload]: addIndex(map)(
                                (record: any, idx: number) => ({
                                    id: idx,
                                    seq: idx,
                                    uploadServerId: record.upload_server_id,
                                    uploadStartTime: record.upload_start_time,
                                    uploadEndTime: record.upload_end_time,
                                    diskPlugTime: record.disk_plug_time,
                                    vehicleId: record.vehicle_id,
                                    xrayUris: record.xray_uris,
                                    uploadStatus: record.upload_status,
                                })
                            )(data.upload_records),
                        },
                    })
                }
            } catch (error) {
                console.error(error)
            }
        },

        async stopUpload(
            payload: {
                serverId: string
                diskId: string
            },
            rootState
        ) {
            try {
                const rv = await request({
                    url: `/data_center/set_disk_finished_manually?upload_server_id=${payload.serverId}&disk_sn=${payload.diskId}`,
                    rootState,
                    dispatch,
                })

                return rv
            } catch (error) {
                console.error(error)
            }
        },

        async changeDiskInfo(
            payload: {
                newRow: DiskType
                params: DiskResponseType
            },
            rootState
        ) {
            const { newRow, params } = payload

            const fixParams = {
                ...params,
                comment: take(MAX_COMMENT_LEN, params.comment || ''),
            }

            const paramsKeys = keys(fixParams)
            const paramStr = reduce(
                (val: string, key: keyof DiskResponseType) => {
                    return `${val}${key}=${fixParams[key]}&`
                },
                ''
            )(paramsKeys)

            try {
                const rv = await request({
                    url: `/disk_management/upsert_disk_info?${paramStr}`,
                    rootState,
                    dispatch,
                })

                if (rv?.code === 0) {
                    const idx = findIndex(
                        propEq('diskId', newRow.diskId),
                        rootState.disk.data
                    )

                    const fixData = update(idx, newRow, rootState.disk.data)

                    dispatch.disk.setData({
                        data: fixData,
                    })

                    return true
                }

                return false
            } catch (error) {
                console.error(error)
            }
        },
    }),
})
