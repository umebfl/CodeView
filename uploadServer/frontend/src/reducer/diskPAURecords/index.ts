import { createModel } from '@rematch/core'

import request from 'src/util/request'

import { RootModel } from '..'
import {
    diskPAURecordsType,
    diskPAURecordsState,
} from 'src/reducer/diskPAURecords/type'

const initState: diskPAURecordsState = {
    data: [],
}

export const diskPAURecords = createModel<RootModel>()({
    state: initState,
    reducers: {
        setData: (state, payload: diskPAURecordsType[]) => {
            return {
                ...state,
                data: payload,
            }
        },
    },
    effects: dispatch => ({
        async initData(_, rootState) {
            const data = await request({
                url: '/data_center/get_diskPAURecords_list',
                rootState,
                dispatch,
            })

            if (data?.diskPAURecords) {
                dispatch.diskPAURecords.setData(data.diskPAURecords)
            }
        },
    }),
})
