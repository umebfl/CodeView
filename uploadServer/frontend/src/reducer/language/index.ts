import { createModel } from '@rematch/core'

import { RootModel } from '..'
import { langSet, languagePropsType } from 'src/reducer/language/type'

const initState: languagePropsType = {
    lang: langSet.zh,
}

export const language = createModel<RootModel>()({
    state: initState,
    reducers: {
        setLang: (state, payload: langSet) => ({
            ...state,
            lang: payload,
        }),
        switchLang: state => ({
            ...state,
            lang: state.lang === langSet.en ? langSet.zh : langSet.en,
        }),
    },
})
