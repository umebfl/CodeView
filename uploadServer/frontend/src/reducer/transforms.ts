import { createTransform } from 'redux-persist'

import { DataSourceEnum } from 'src/reducer/userConfig/type'

const SetTransform = createTransform(
    // transform state on its way to being serialized and persisted.
    (inboundState: any, key) => {
        // old persisted state
        if (
            inboundState.dataSource &&
            (inboundState.dataSource === 'guangzhou' ||
                inboundState.dataSource === 'shenzhen')
        ) {
            return {
                ...inboundState,
                dataSource:
                    inboundState.dataSource === 'guangzhou'
                        ? DataSourceEnum.guangzhou
                        : DataSourceEnum.shenzhen,
            }
        }

        return inboundState
    },

    // transform state being rehydrated
    (outboundState: any, key) => {
        // old persisted state
        if (
            outboundState.dataSource &&
            (outboundState.dataSource == 'guangzhou' ||
                outboundState.dataSource == 'shenzhen')
        ) {
            return {
                ...outboundState,
                dataSource:
                    outboundState.dataSource === 'guangzhou'
                        ? DataSourceEnum.guangzhou
                        : DataSourceEnum.shenzhen,
            }
        }

        // convert mySet back to a Set.
        return outboundState
    },
    // define which reducers this transform gets called for.
    // { whitelist: ['someReducer'] }
    { whitelist: ['userConfig', 'language'] }
)

export default SetTransform
