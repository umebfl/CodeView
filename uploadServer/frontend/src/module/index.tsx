/** No need unit test */
import React from 'react'
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import { useSelector } from 'react-redux'
import Box from '@mui/material/Box'
import { useTheme } from '@mui/material/styles'

import HomePage from 'src/module/homePage'
import UploadServer, {
    UploadServerList,
    UploadServerDetail,
} from 'src/module/uploadServer'
import Disk, { DiskList, DiskPlugRecords, DiskUploadLog } from 'src/module/disk'

import NoFound from 'src/module/noFound'
import { RootState } from 'src/reducer/type'

// TODO: google analysis
function Router() {
    const theme = useTheme()
    const { rehydrated } = useSelector((state: RootState) => state._persist)

    return (
        <Box
            sx={{
                display: 'flex',
                flex: 1,
                background: theme.color.grey2,
            }}
        >
            {rehydrated && (
                <BrowserRouter>
                    <Routes>
                        <Route path="/" element={<HomePage />}>
                            <Route path="up" element={<UploadServer />}>
                                <Route index element={<UploadServerList />} />
                                <Route
                                    path="detail/:id"
                                    element={<UploadServerDetail />}
                                />
                            </Route>
                            <Route path="disk" element={<Disk />}>
                                <Route index element={<DiskList />} />
                                <Route
                                    path="plugRecords"
                                    element={<DiskPlugRecords />}
                                />
                                <Route
                                    path="uploadLog"
                                    element={<DiskUploadLog />}
                                />
                            </Route>
                            <Route path="*" element={<NoFound />} />
                        </Route>
                    </Routes>
                </BrowserRouter>
            )}
        </Box>
    )
}

export default Router
