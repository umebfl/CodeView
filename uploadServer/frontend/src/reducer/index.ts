/** No need unit test */
import { Models } from '@rematch/core'

import { uploadServer } from 'src/reducer/uploadServer'
import { globalLoading } from 'src/reducer/globalLoading'
import { snackbar } from 'src/reducer/snackbar'
import { userConfig } from 'src/reducer/userConfig'
import { language } from 'src/reducer/language'

export interface RootModel extends Models<RootModel> {
    uploadServer: typeof uploadServer
    globalLoading: typeof globalLoading
    snackbar: typeof snackbar
    userConfig: typeof userConfig
    language: typeof language
}

export const models: RootModel = {
    uploadServer,
    globalLoading,
    snackbar,
    userConfig,
    language,
}
