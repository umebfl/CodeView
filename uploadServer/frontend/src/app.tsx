import React, { FC } from 'react'

import { Provider } from 'react-redux'
import { ThemeProvider } from '@mui/material/styles'

import DefaultTheme from 'src/theme'
import Router from 'src/module'
import { store } from 'src/reducer/store'

import SnackbarContainer from 'src/component/snackbar'
import { Store } from 'src/reducer/type'

import './app.css'

interface ContextType {
    children: any
    initStore?: Store
}

export const Context = ({ children, initStore }: ContextType) => (
    <Provider store={initStore || store}>
        <ThemeProvider theme={DefaultTheme}>
            {children}
            <SnackbarContainer />
        </ThemeProvider>
    </Provider>
)

const App: FC = () => {
    return (
        <Context>
            <Router />
        </Context>
    )
}

export default App
