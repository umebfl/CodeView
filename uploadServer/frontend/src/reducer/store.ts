/** No need unit test */
import { init } from '@rematch/core'
import createRematchPersist from '@rematch/persist'
import AsyncStorage from '@react-native-community/async-storage'

import { models, RootModel } from 'src/reducer'

const persistPlugin = createRematchPersist<RootModel, any>({
    key: 'autox',
    storage: AsyncStorage,
    whitelist: ['userConfig', 'language'],
    throttle: 1000,
    debug: false,
    version: 6,
})

export const getInitStore = () =>
    init({
        models,
        plugins: [persistPlugin],
    })

export const store = getInitStore()
