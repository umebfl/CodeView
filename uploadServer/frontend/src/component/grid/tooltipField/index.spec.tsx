import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'

import TooltipField from 'src/component/grid/tooltipField'
import { Context } from 'src/app'
import { getInitStore } from 'src/reducer/store'

describe('tooltipField', () => {
    it('renders correctly', () => {
        const tree = render(
            <Context initStore={getInitStore()}>
                <TooltipField title="test" />
            </Context>
        )

        expect(tree).toMatchSnapshot()
    })
})
