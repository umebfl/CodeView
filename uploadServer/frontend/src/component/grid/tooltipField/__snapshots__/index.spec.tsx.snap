// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tooltipField renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1afp10f"
      >
        <div
          aria-label="test"
          class="MuiBox-root css-746fct"
          data-mui-internal-clone-element="true"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1afp10f"
    >
      <div
        aria-label="test"
        class="MuiBox-root css-746fct"
        data-mui-internal-clone-element="true"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`tooltipField should render UI  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1afp10f"
      >
        <div
          aria-label="test"
          class="MuiBox-root css-746fct"
          data-mui-internal-clone-element="true"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1afp10f"
    >
      <div
        aria-label="test"
        class="MuiBox-root css-746fct"
        data-mui-internal-clone-element="true"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`tooltipField should render UI 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1afp10f"
      >
        <div
          aria-label="test"
          class="MuiBox-root css-746fct"
          data-mui-internal-clone-element="true"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1afp10f"
    >
      <div
        aria-label="test"
        class="MuiBox-root css-746fct"
        data-mui-internal-clone-element="true"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`tooltipField should render hints  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1afp10f"
      >
        <div
          aria-label="test"
          class="MuiBox-root css-746fct"
          data-mui-internal-clone-element="true"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1afp10f"
    >
      <div
        aria-label="test"
        class="MuiBox-root css-746fct"
        data-mui-internal-clone-element="true"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
